includes/fbbot/class-omise-fbbot-fbpage-setup.php

```diff
@@ -0,0 +1,151 @@
+			error_log( print_r( $body->error, true ) );
+		} else {
+			error_log("Update : set_page_get_started_button success");
+		}
+	}
+
+	public function set_page_greeting_message() {
+		$blog_title = get_bloginfo('name');
+
+		$data = array(
+			array(
+			  'locale' => 'default',
+			  'text' => "Hi {{user_first_name}}, Welcome to " . $blog_title . "\r\nYou can . . . \r\n- Shopping feature and cool products\r\n- Check your order status \r\n- and more :D"
+			)
+		  );
+
+		$url = Omise_FBBot_Configurator::get_fb_profile_endpoint();
+
+		$response = Omise_FBBot_HTTPService::send_request( $url, $data );
+
+		$body = json_decode( $response['body'] );
+		if ( isset( $body->error ) ) {
+			error_log( print_r( $body->error, true ) );
+		} else {
+			error_log("Update : set_page_greeting_message success");
+		}
+	}
+
+	public function set_page_persistent_menu() {
+		// clear old persistent menu
+		$delete_persistent_menu_data = array(
+			'fields' => array(
+				'persistent_menu'
+			  )
+		  );
+
+		$url = Omise_FBBot_Configurator::get_fb_profile_endpoint();
+
+		$deleteResponse = Omise_FBBot_HTTPService::send_delete_request( $url, $delete_persistent_menu_data );
+		$body = json_decode( $deleteResponse['body'] );
+		if ( isset( $body->error ) ) {
+			error_log( print_r( $body->error, true ) );
+		} else {
+			error_log("Clear : old persistent_menu success");
+		}
+
+		// Update new persistent menu
+		$persistent_menu_data = array(
+		'persistent_menu' => array(
+			array(
+			'locale' => 'default',
+			'composer_input_disabled' => false,
+			'call_to_actions' => array(
+				array(
+					'type' => 'postback',
+					'title' => 'Check order status',
+					'payload' => Omise_FBBot_Payload::CHECK_ORDER
+					),
+
+				array(
+					'type' => 'web_url',
+					'title' => 'View Website',
+					'url' => site_url()
+					),
+
+				array(
+					'type' => 'postback',
+					'title' => 'Help',
+					'payload' => Omise_FBBot_Payload::HELP
+					)
+				
+				)
+			)
+		  )
+		);
+
+		$response = Omise_FBBot_HTTPService::send_request( $url, $persistent_menu_data );
+		$body = json_decode( $response['body'] );
+
+		if ( isset( $body->error ) ) {
+			error_log( print_r( $body->error, true ) );
+		} else {
+			error_log("Update : setup persistent_menu success");
+		}
+
+	}
+
+	public function set_page_whitelist_domain() {
+		$data = array(
+			'whitelisted_domains' => array( site_url() )
+			);
+
+		$url = Omise_FBBot_Configurator::get_fb_profile_endpoint();
+
+		$response = Omise_FBBot_HTTPService::send_request( $url, $data );
+
+		$body = json_decode( $response['body'] );
+		if ( isset( $body->error ) ) {
+			error_log( print_r( $body->error, true ) );
+		} else {
+			error_log("Update : whitelist_domain success");
+		}
+	}
+
+	public function facebook_page_setup() {
+		if ( ! $this->facebook_page_access_token ) {
+			error_log( 'Facebook page access token is empty, bot will setup page later' );
+			return;
+		}
+
+		self::$instance->set_page_get_started_button();
```

Comment:
@igroomgrim ✋ wonder, why don't we just post everything in one time? (because all of these 4 methods are posting to the same endpoint).

Reply:
Same endpoint but different data. I see they have different key of data but not sure if we can merge it and sent to endpoint in 1 time

Reply:
@igroomgrim fyi, just tested, it can. In my refactoring code, I tested with:
```php
$response = wp_safe_remote_post(
	$this->get_facebook_profile_endpoint(),
	array(
		'timeout' => 60,
		'body'    => array(
			'get_started'         => array( 'payload' => 'GET_START_CLICKED' ),
			'greeting'            => array( array( 'locale' => 'default', 'text' => 'Hi Guzzilar') ),
			'whitelisted_domains' => array( site_url(), 'https://www.omise.co' )
		)
	)
);
```